plugins/analysis-kuromoji/src/main/java/org/opensearch/plugin/analysis/kuromoji/AnalysisKuromojiPlugin.java

```diff
@@ -80,6 +83,9 @@ public Map<String, AnalysisProvider<TokenizerFactory>> getTokenizers() {
 
     @Override
     public Map<String, AnalysisProvider<AnalyzerProvider<? extends Analyzer>>> getAnalyzers() {
-        return singletonMap("kuromoji", KuromojiAnalyzerProvider::new);
+        Map<String, AnalysisProvider<AnalyzerProvider<? extends Analyzer>>> extra = new HashMap<>();
+        extra.put("kuromoji", KuromojiAnalyzerProvider::new);
+        extra.put("kuromoji_completion", KuromojiCompletionAnalyzerProvider::new);
```

Comment:
The kuromoji_completion analyser has been created for use with completion fields.

Whenever a user wants to use it in a completion field, the analyzer for INDEX and QUERY must be created in settings.index.analysis as shown below.
```
{
  "mappings": {
    "properties": {
      "suggest_ja": {
        "type": "completion",
        "analyzer": "japanese_completion_index",
        "search_analyzer": "japanese_completion_query"
      }
    }
  },
  "settings": {
    "index": {
      "analysis": {
        "analyzer": {
          "japanese_completion_index": {
            "mode": "index",
            "type": "kuromoji_completion"
          },
          "japanese_completion_query": {
            "mode": "query",
            "type": "kuromoji_completion"
          }
        }
      }
    }
  }
}
```

Should we create a KuromojiCompletionIndexAnalyzerProvider(kuromoji_completion_index) and a KuromojiCompletionQueryAnalyzerProvider(kuromoji_completion_query) so that all users can use them by default without settings.index.analysis configuration?

Reply:
I was unsure which option to choose, but I think it's fine to leave things as they are now. Please let me know if you have any feedback.

Reply:
I agree that the current setup is a good starting point. 

If people want the providers, they can always be added later. :+1: